routes/project/project_api.js

```diff
@@ -23,22 +23,30 @@ router.post('/saveProject', function (req, res, next) {
             return;
         }
         if(req.body.id){
-            project_db.editProject(req.body, function (err) {
-                if (err) {
-                    console.log(err);
-                    res.status(400).json({msg: 'Failed to save'});
-                    return;
+            let hasErr = false;
+            project_db.editProject(req.body).catch(err => {
+                hasErr = true;
+            }).then(() => {
+                if(hasErr){
+                    res.status(400).json({msg: 'Failed to create'});
+                }
+                else{
+                    res.json({})
                 }
-                res.json({});
             });
         }else{
-            project_db.createProject(req.body, function (err, pid) {
-                if (err) {
-                    console.log(err);
+            var hasErr = false;
+            var pid = -1;
+            project_db.createProject(req.body).catch(err => {
+                hasErr = true;
+            }).then(result => {
+                pid = result;
+                if(hasErr){
                     res.status(400).json({msg: 'Failed to create'});
-                    return;
                 }
-                res.json({pid: pid});
+                else{
+                    res.json({pid: pid})
```

Comment:
Change `result` to `pid` and directly retrun it instead of assigning `result` to `pid` and then return pid.

Reply:
Combine this with last 3 comments